tests/test_context.py

```diff
@@ -166,3 +172,93 @@ def test_copy_to_frontend():
                     os.listdir(os.path.join(temp_dir, rec_folder)) ==
                     os.listdir(os.path.join(temp_dir_2, rec_folder))
             )
+
+
+class TestPerRunDefaults(unittest.TestCase):
+    """Test the per-run defaults options of a context"""
+    def setUp(self):
+        """
+        Get a folder to write (Similar to tempfile.TemporaryDirectory) to get
+        a temp folder in the temp dir. Cleanup after the tests.
+        """
+        temp_folder = uuid.uuid4().hex
+        self.tempdir = os.path.join(tempfile.gettempdir(), temp_folder)
+        assert not os.path.exists(self.tempdir)
+
+    def tearDown(self):
+        shutil.rmtree(self.tempdir)
+
+    def test_register_no_defaults(self, runs_default_allowed=False):
+        """Test if we only register a plugin with no run-defaults"""
+        st = self.get_context(runs_default_allowed)
+        assert not self._has_per_run_default(Records)
+        st.register(Records)
+        st.select_runs()
+
+    def test_register_no_defaults_but_allowed(self):
+        self.test_register_no_defaults(runs_default_allowed=True)
+
+    def test_register_with_defaults(self, runs_default_allowed=False):
+        """Test if we register a plugin WITH run-defaults"""
+        st = self.get_context(runs_default_allowed)
+        assert not self._has_per_run_default(Records)
+        assert self._has_per_run_default(Peaks)
+        st.register(Records)
+        if not runs_default_allowed:
+            self.assertRaises(strax.InvalidConfiguration, st.register, Peaks)
+        else:
+            st.select_runs()
+
+    def test_register_with_defaults_and_allowed(self):
+        self.test_register_with_defaults(runs_default_allowed=True)
+
+    def test_register_all_no_defaults(self, runs_default_allowed=False):

```

Comment:
Sorry, but could you explain me the difference to `test_register_with_defaults`  I cannot see any.

Reply:
yes, I just wanted to to check the `st.register_all(strax.testutils)` function. Such that this is affected treated similarly to the `st.register` function

Reply:
Ah okay, thanks. I am not sure if we gain a lot by testing st.register_all too. Under the hood it calls st.register anyway.

Reply:
I agree, I wondered about it, so looked it up and convinced myself that this test was probably not needed. However, it was easy to add and the tests completes within <seconds, so it does not hurt either.